File: backend/app/api/routes/analytics.py
from fastapi import APIRouter

from app.api.deps import SessionDep
from app.schemas.analytics import FunnelData, SankeyData, StatusEventRead
from app.services import analytics_service

router = APIRouter(prefix="/analytics", tags=["analytics"])


@router.get("/sankey", response_model=SankeyData)
def get_sankey(session: SessionDep):
    return analytics_service.get_sankey(session)


@router.get("/funnel", response_model=FunnelData)
def get_funnel(session: SessionDep):
    return analytics_service.get_funnel(session)


@router.get("/timeline/{application_id}", response_model=list[StatusEventRead])
def get_timeline(application_id: int, session: SessionDep):
    return analytics_service.get_timeline(session, application_id)

File: frontend/src/api/analytics.ts
import { get } from "./client";
import type { FunnelData, SankeyData, StatusEventRead } from "../types";

export const getSankey = () => get<SankeyData>("/analytics/sankey");
export const getFunnel = () => get<FunnelData>("/analytics/funnel");
export const getTimeline = (applicationId: number) =>
  get<StatusEventRead[]>(`/analytics/timeline/${applicationId}`);
